packages/pieces/sftp/src/lib/actions/create-file.ts

```diff
@@ -0,0 +1,94 @@
+import { createAction, Property } from "@activepieces/pieces-framework";
+import { httpClient, HttpMethod } from "@activepieces/pieces-common";
+import Client from 'ssh2-sftp-client'
+
+export const createFile = createAction({
+	name: 'create_file', 
+    displayName:'Create new file',
+    description: 'Create a new file in the given path',
+	props: {
+		fileName: Property.ShortText({
+			displayName: 'File name',
+			description: 'The name of the file to create',
+			required: true,
+		}),
+        fileContent: Property.LongText({
+            displayName: 'File content',
+            description: 'The content of the file to create',
+            required: true,
+        }),
+        // host, port, username, password
+        host: Property.ShortText({
```

Comment:
Create Custom Authentication Property and put host, username, post, password inside it.